sanic/pages/error.py

```diff
@@ -0,0 +1,101 @@
+from typing import Any, Mapping
+
+import tracerite.html
+
+from html5tagger import E
+from tracerite import html_traceback, inspector
+
+from sanic.request import Request
+
+from .base import BasePage
+
+
+# Avoid showing the request in the traceback variable inspectors
+inspector.blacklist_types += (Request,)
+
+
+class ErrorPage(BasePage):
+    STYLE_APPEND = tracerite.html.style
+
+    def __init__(
+        self,
+        title: str,
+        text: str,
+        request: Request,
+        exc: Exception,
+        full: bool,
+    ) -> None:
+        super().__init__()
+        # Internal server errors come with the text of the exception, which we don't want to show to the user.
+        # FIXME: This needs to be done some place else but I am not digging into that now.
+        if "Internal Server Error" in title:
+            text = "The application encountered an unexpected error and could not continue."
+        self.TITLE = E.strong(request.app.name)(" cannot handle your request")
+        self.title = title
+        self.text = text
+        self.request = request
+        self.exc = exc
+        self.full = full
+
+    def _head(self) -> None:
+        self.doc._script(tracerite.html.javascript)
+        super()._head()
+
+    def _body(self) -> None:
+        debug = self.request.app.debug
+        try:
+            route_name = self.request.route.name
+        except AttributeError:
+            route_name = "[route not found]"
+        with self.doc.main:
+            self.doc.h1(f"⚠️ {self.title}").p(self.text)
+            # Show context details if available on the exception
+            context = getattr(self.exc, "context", None)
+            if context:
+                self._key_value_table(
+                    "Issue context", "exception-context", context
+                )
+
+            if not debug:
+                return
+            # Show additional details in debug mode,
+            # open by default for 500 errors
+            with self.doc.details(open=self.full, class_="smalltext"):
```

Comment:
If I am reading this right, I think we may be using `full` here incorrectly. If this is meant to be full/minimal from the error page renderers, then it should be doing what you are doing with debug here and not controlling whether the details are open or not.

Reply:
https://sanic.dev/en/guide/best-practices/exceptions.html#html

Reply:
The intent is different than debug. The details are supposed to be open by default for 500 errors and not for 404 (roughly speaking). `full` accomplishes that, but if you have a better variable that does the same, let me know.

Reply:
Right, my point is that its misusing something already. You shouldn't read debug from the app instance and should use full for that purpose.

For deciding if it's open or not, I would use `getattr(exc, "quiet", False)`.